src/plugins/intel_gpu/src/runtime/ocl/ocl_memory.cpp

```diff
@@ -475,19 +477,23 @@ gpu_usm::gpu_usm(ocl_engine* engine, const layout& layout, allocation_type type)
     , memory(engine, layout, type, nullptr)
     , _buffer(engine->get_usm_helper())
     , _host_buffer(engine->get_usm_helper()) {
+    std::vector<cl_mem_properties_intel> properties;
+    if (engine->get_device_info().vendor_id == INTEL_VENDOR_ID) {
+        properties = {CL_MEM_FLAGS, CL_MEM_ALLOW_UNRESTRICTED_SIZE_INTEL, 0};
+    }
+
     switch (get_allocation_type()) {
     case allocation_type::usm_host:
-        _buffer.allocateHost(_bytes_count);
+        _buffer.allocateHost(_bytes_count, &properties[0]);
```

Comment:
Accessing an empty vector (in case of engine->get_device_info().vendor_id == INTEL_VENDOR_ID) at index 0 with operator[] could cause UB probably
And specs say (regarding the properties argument): _The list is terminated with the special property 0_. So it should be either nullptr or vector with single 0 value by default